src/service/cia301/co_ssdo.c

```diff
@@ -372,7 +372,7 @@ CO_ERR COSdoInitUploadSegmented(CO_SDO *srv, uint32_t size)
     /* setup SDO buffer */
     srv->Buf.Cur  = srv->Buf.Start;
     srv->Buf.Num  = 0;
-    result = COObjRdBufStart(srv->Obj, srv->Node, srv->Buf.Cur, 0);
+    result = COObjRdBufStart(srv->Obj, srv->Node, srv->Buf.Cur, size);
```

Comment:
Why do you think, the `0` should be replaced by `size` in the transfer initialization? Is this change coming from a "debugging session"?

(With this change, the integration tests for checking the segmented upload are failed.)

Reply:
Maybe we should improve this area with a comment to highlight the idea of this line:

```c
/* Ensure next reading with `COObjRdBufCont()` from start of object entry.
 * Read no actual data for the response of 'SDO upload initiate'.
 */
```

Reply:
Hi @michael-hillmann,

> Why do you think, the `0` should be replaced by `size` in the transfer initialization? Is this change coming from a "debugging session"?

Yes, while debugging a segmented transfer(UNSIGNED48) SDO fails with size hardcoded to 0, because in integer48 type I've implemented, if the size requested is different from 6 bytes(the size of integer48) I return an error like in the other basic types. But it looks good to me because the entire value(integer48) must be read at the beginning and placed into 'srv->Buf.Start' and then later the stack keeps answering different segments.

By the way, I think it would be better to use 'srv->Buf.Start' pointer instead of 'srv->Buf.Cur'. It's more clear. What also about this?

Reply:
Ok, you are right. There is an issue when using segmented transfers for the basic data types. With some investigation, I realize a solution which breaks no other test and documents our intend:

We need to change the common function 'read from start of buffer' to check the required behavior of SDO segmented transfer. This is resetting the read index without reading any bytes:

```c
CO_ERR COObjRdBufStart(struct CO_OBJ_T *obj, struct CO_NODE_T *node, uint8_t *buffer, uint32_t size)
{
    const CO_OBJ_TYPE *type;
    CO_ERR result = CO_ERR_OBJ_ACC;

    ASSERT_PTR_ERR(obj,       CO_ERR_BAD_ARG);
    ASSERT_PTR_ERR(obj->Type, CO_ERR_BAD_ARG);
    ASSERT_PTR_ERR(node,      CO_ERR_BAD_ARG);
    ASSERT_PTR_ERR(buffer,    CO_ERR_BAD_ARG);

    type = obj->Type;
    if (type->Read != NULL) {
        /* mandatory: reset object */
        (void)COObjReset(obj, node, 0);

        /* optional: read from object */
        if (size > 0) {
            result = type->Read(obj, node, (void *)buffer, size);
        } else {
            result = CO_ERR_NONE;
        }
    }
    return (result);
}
```

Same applies to `COObjWrBufStart()`.

Reply:
I've just tested this by modifying both COObjRdBufStart() and COObjWrBufStart() and it works well.
Do you want me to write such patch or you do that?

Reply:
If you include it in this PR we have a reference, why we change it. Would be nice.

Reply:
Sure. I'll add this patch tonight and push --force. So you will have 4 patches on this PR.